pstress/repeat_crash.sh

```diff
@@ -0,0 +1,227 @@
+}
+
+echoit "========================================================="
+echoit "Incident Directory: $incident_directory";
+echoit "Nth step: $step";
+echoit "Mode previous|current|after: $mode";
+echoit "Base directory: $basedir";
+echoit "========================================================="
+
+RANDOM=`date +%s%N | cut -b14-19`; RANDOMD=$(echo $RANDOM$RANDOM$RANDOM | sed 's/..\(......\).*/\1/')
+mkdir $RANDOMD
+workdir=`pwd`/$RANDOMD
+SCRIPT_PWD=$(cd `dirname $0` && pwd)
+CONFIGURATION_FILE=pstress-run.conf
+source $incident_directory/$CONFIGURATION_FILE
+SEED=$(<$incident_directory/seed)
+
+
+if [ ! -d "$incident_directory" ]; then
+  echo "Error: ${incident_directory} not found. Can not continue."
+  exit 1
+fi
+
+if [ ! -d "$incident_directory/$step" ]; then
+  echo "Error: Invalid step=$step. Can not continue."
+  exit 1
+fi
+
+if [ ! -d "$basedir" ]; then
+  echo "Error: Invalid basedir=$basedir. Can not continue."
+  exit 1
+fi
+
+if [[ "$mode" == "current" ]]; then
+  TRIAL=$[ $step - 1 ]
+  if [ $TRIAL -le 0 ]; then
+    echo "The value of either mode or nth step is incorrect."
+    echo "Cannot continue. Exiting"
+    exit
+  else
+    echoit "Taking backup of datadir from $incident_directory/$TRIAL/data into $workdir/$TRIAL/data"
+    mkdir $workdir/$TRIAL
+    rsync -ar --exclude='*core*' $incident_directory/$TRIAL/data/ $workdir/$TRIAL/data 2>&1
+    echoit "Copying step_$TRIAL.dll file into $workdir"
+    cp $incident_directory/step_$TRIAL.dll $workdir
+  fi
+elif [[ "$mode" == "previous" ]]; then
+  TRIAL=$[ $step - 2 ]
+  if [ $TRIAL -le 0 ]; then
+    echo "The value of either mode or nth step is incorrect."
+    echo "Cannot continue. Exiting"
+    exit
+  else
+    echoit "Taking backup of datadir from $incident_directory/$TRIAL/data into $workdir/$TRIAL/data"
+    mkdir $workdir/$TRIAL
+    rsync -ar --exclude='*core*' $incident_directory/$TRIAL/data/ $workdir/$TRIAL/data 2>&1
+    echoit "Copying step_$TRIAL.dll file into $workdir"
+    cp $incident_directory/step_$TRIAL.dll $workdir
+  fi
+elif [[ "$mode" == "after" ]]; then
+  TRIAL=$step
+  echoit "Taking backup of datadir from $incident_directory/$TRIAL/data into $workdir/$TRIAL/data"
+  mkdir $workdir/$TRIAL
+  rsync -ar --exclude='*core*' $incident_directory/$TRIAL/data/ $workdir/$TRIAL/data 2>&1
+  echoit "Copying step_$TRIAL.dll file into $workdir"
+  cp $incident_directory/step_$TRIAL.dll $workdir
+else
+  echo "Error: Invalid mode=$mode. Can not continue."
+  exit 1
+fi
+
+myextra=$(<$incident_directory/$step/MYEXTRA)
+mysqld_start_timeout=30
+port=50661
+socket=/tmp/mysql_50661.sock
+
+# Kill the server
+kill_server(){
+  SIG=$1
+  echoit "Killing the server with Signal $SIG";
+  { kill -$SIG ${MPID} && wait ${MPID}; } 2>/dev/null
+}
+
+repeat_crash() {
+
+TRIAL=$[ $TRIAL + 1 ]
+echoit "====== TRIAL #$TRIAL ======"
+echoit "Ensuring there are no relevant mysqld server running"
+KILLPID=$(ps -ef | grep "mysqld" | grep -v grep | awk '{print $2}' | tr '\n' ' ')
```

Comment:
only kill mysqld started with the same user and with repeat_crash script.  It might be some other user are running some test.

Reply:
The script cannot kill any process started by a different user due to permissions check. I have tested this.

Reply:
To double check I will grep for the incident directory path as well, so that we kill unique mysqld